symbolic/src/Data/Macaw/Symbolic/Stack.hs

```diff
@@ -0,0 +1,115 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ImplicitParams #-}
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Data.Macaw.Symbolic.Stack
+  ( stackArray
+  , mallocStack
+  , ExtraStackSlots(..)
+  , ArrayStack(..)
+  , createArrayStack
+  ) where
+
+import Data.BitVector.Sized qualified as BVS
+import Data.Parameterized.Context as Ctx
+import Lang.Crucible.Backend qualified as C
+import Lang.Crucible.LLVM.DataLayout qualified as CLD
+import Lang.Crucible.LLVM.MemModel qualified as CLM
+import What4.Interface qualified as WI
+import What4.Symbol qualified as WSym
+
+-- | Helper, not exported
+stackAlign :: CLD.Alignment
+stackAlign = CLD.noAlignment
+
+-- | Create an SMT array representing the program stack.
+--
+-- Builds a 'WI.freshConstant' with name "stack_array".
```

Comment:
These are rather low-level implementation details. I think that for the sake of Haddocks, it would suffice to say that you're creating a fresh SMT array.